src/ServerConfig_TEST.cc

```diff
@@ -228,3 +228,27 @@ TEST(ServerConfig, GenerateRecordPlugin)
   EXPECT_EQ(plugin.Name(), "ignition::gazebo::systems::LogRecord");
 }
 
+//////////////////////////////////////////////////
+TEST(ServerConfig, SdfRoot)
+{
+  ServerConfig config;
+  EXPECT_FALSE(config.SdfRoot());
+  EXPECT_TRUE(config.SdfFile().empty());
+  EXPECT_TRUE(config.SdfString().empty());
+
+  config.SetSdfString("string");
+  EXPECT_FALSE(config.SdfRoot());
+  EXPECT_TRUE(config.SdfFile().empty());
+  EXPECT_FALSE(config.SdfString().empty());
+
+  config.SetSdfFile("file");
+  EXPECT_FALSE(config.SdfRoot());
+  EXPECT_FALSE(config.SdfFile().empty());
+  EXPECT_TRUE(config.SdfString().empty());
+
+  sdf::Root root;
+  config.SetSdfRoot(root);
+  EXPECT_TRUE(config.SdfRoot());
+  EXPECT_TRUE(config.SdfFile().empty());
+  EXPECT_TRUE(config.SdfString().empty());
```

Comment:
Add assertions for SourceType

Reply:
Added in [7d33cf0](https://github.com/ignitionrobotics/ign-gazebo/pull/1333/commits/7d33cf0aafacefa3a4bdaa1522ce02b7d7b5263b)